src/coreclr/vm/callstubgenerator.cpp

```diff
@@ -679,6 +686,12 @@ CallStubHeader *CallStubGenerator::GenerateCallStub(MethodDesc *pMD, AllocMemTra
     {
         TypeHandle thReturnValueType;
         CorElementType thReturnType = sig.GetReturnTypeNormalized(&thReturnValueType);
+        if (isSpecialConstructor)
+        {
+            assert(thReturnType == ELEMENT_TYPE_VOID);
+            // FIXME: String?
+            thReturnType = ELEMENT_TYPE_OBJECT;
```

Comment:
```suggestion
            thReturnType = ELEMENT_TYPE_CLASS;
```

Normally, I defer to @jkotas, the only time we care about `ELEMENT_TYPE_OBJECT` is when working directly with signatures otehrwise we use `ELEMENT_TYPE_CLASS`. I don't think it matters in this case, but that is what I recall.